src/lower.c

```diff
@@ -2278,9 +2422,16 @@ static vcode_reg_t lower_fcall(tree_t fcall)
 {
    tree_t decl = tree_ref(fcall);
 
+   if (cover_enabled(cover_tags, COVER_MASK_EXPRESSION))
```

Comment:
I think these scopes will only ever have two tags in them (perhaps we should call them coverage bins instead). Could we instead just number expressions sequentially inside the parent scope?

Reply:
Yes, that is a good idea. The original idea of "scope for everything" which targeted independce
of coverage hierarchy between different parts of large `case` statements representing complex
FSM falls apart here. If someone would have way too complex / nested logic expression
representing some combo logic, its anyway good coding practice to create sub-signals and
split such expression to multiple stand-alone expressions, each nested under separate
signal assignment.